packages/blade/src/components/Switch/AnimatedThumbIcon.native.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import type { EasingFunctionFactory } from 'react-native-reanimated';
+import Animated, {
+  interpolate,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+
+import { useTheme } from '~components/BladeProvider';
+import type { SvgProps } from '~components/Icons/_Svg/Svg/types';
+import Svg from '~components/Icons/_Svg';
+
+const AnimatedThumbIcon = ({
+  children,
+  isChecked,
+  width,
+  height,
+  viewBox,
+  fill,
+}: {
+  children: React.ReactNode;
+  isChecked: boolean;
+} & SvgProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const opacity = useSharedValue(isChecked ? 1 : 0);
+
+  React.useEffect(() => {
+    opacity.value = withTiming(isChecked ? 1 : 0, {
+      duration: theme.motion.duration.xquick,
+      easing: theme.motion.easing.standard.effective as EasingFunctionFactory,
+    });
+  }, [isChecked, opacity]);
```

Comment:
I see you didn't add `translateX` in deps for AnimatedThumb but you've added `opacity` in deps for this. Any reason why?

Reply:
Yeah right I don't think its needed to add this in dependency these shared values works like refs. Otherwise this would've gotten into a infinite loop right now (setting opacity in effect and then adding opacity in the effect's deps)